src/modules/qt/qimage_wrapper.cpp

```diff
@@ -255,21 +265,42 @@ void refresh_image( producer_qimage self, mlt_frame frame, mlt_image_format form
 		QImage scaled = interp? qimage->scaled( QSize( width, height ), Qt::IgnoreAspectRatio, Qt::SmoothTransformation ) :
 			qimage->scaled( QSize(width, height) );
 
-		// Convert scaled image to target format (it might be premultiplied after scaling).
-		scaled = scaled.convertToFormat( qimageFormat );
-
 		// Store width and height
 		self->current_width = width;
 		self->current_height = height;
 
 		// Allocate/define image
-		self->format = has_alpha ? mlt_image_rgb24a : mlt_image_rgb24;
-		int image_size = mlt_image_format_size( self->format, self->current_width, self->current_height, NULL );
-		self->current_image = ( uint8_t * )mlt_pool_alloc( image_size );
 		self->current_alpha = NULL;
 		self->alpha_size = 0;
 
 		// Copy the image
+		int image_size;
+#if QT_VERSION >= 0x050200
+		if ( has_alpha )
+		{
+			image_size = 4 * width * height;
+			self->format = mlt_image_rgb24a;
+			scaled = scaled.convertToFormat( QImage::Format_RGBA8888 );
+		}
+		else
+		{
+			image_size = 3 * width * height;
+			self->format = mlt_image_rgb24;
+#if QT_VERSION >= 0x051300
+			scaled.convertTo( QImage::Format_RGB888 );
```

Comment:
I realize the difference between `convertTo()` and `convertToFormat()`, but does it provide any advantage? Otherwise, it clutters the code, and the block just above this does not have the build option to use `convertTo()`.

Reply:
@j-b-m good you removed this check: 5.13 should have been 0x050D00 (or better QT_VERSION_CHECK(5,13,0)